### PR TITLE
S5 t3/implement cancellation logic

### PR DESCRIPTION
**Pull request by catai9**
**Scrum Board Ticket:** S5 T3: Implement worker cancellation logic for the back end

**Story:** Story 5 Cancel specific day 

**Description:** Updated the cancellation logic for a worker on a specific date. 

**Testing Criteria (how I tested the code):** 
1. Running locally and testing the `/api/cancel-specific-day` endpoints on port 4000 using Postman with various combinations of  workerId and specificDate to ensure that the response was as expected and cross-referenced with the values in my local database. 
- workerId = 8000000 and specificDate = 2020-12-20 => true returned and relevant appointments marked as cancelled in local database table. 
- no values specified for workerId or specificDate => error thrown 
- invalid specificDate format, e.g., 'hello' => error thrown
- workerId is string => throw error

2. Running the automated server test codes and ensured that there was 100% file coverage (and passing tests) for all the unit tests. Please note that `account.js` has slightly lower than 100% file coverage and is being addressed in another ticket.
3. Running the unit tests in the front end and ensured 100% file coverage (and passing tests) for the `AppointmentList.js`, `LogIn.js` and `CheckboxApplication.js` components. The other tests are not set up yet (in the scope of another ticket) and thus the low file coverage for the other front end files is expected.

**How Reviewers can Test Code:**
1. Run the code locally and ensure no complications when starting up (run the command `npm run-script start:dev` in the root folder OR run the command `npm start` if your node and npm versions are above 12.10.0 for node and 6.10.3 for npm).
a. Open Postman and test the endpoint by sending requests to the `api/cancel-specific-day` endpoint on port 4000 with various combinations of query parameters `workerId` and `specificDate` to ensure that the response is as expected for numerical values and string values (refer to above list for examples).
**_Make sure to specify an `accessToken` as this is a private endpoint. An example of an accessToken is `XcCa92ZvOnQKZsGtOKOa`._**

2. Run the automated tests locally and ensure that all tests pass and have 100% file coverage. 
a. Navigate inside the `server` folder and run `npm test`. 
b. Navigate inside the `react-ui` folder and run `npm test`. All tests should pass and for the `AppointmentList.js`, `LogIn.js` and `CheckboxApplication.js` components should have 100% file coverage (the rest are not set up yet and are in the scope for another ticket).

### CODE REVIEW

File: server/models/db/workerTimeslot.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 const _ = require('lodash');
 
 const { db } = require('../../lib/connection');
+
 const Tables = require('../../constants/tables.json');
-const TimeslotStatus  = require('../../constants/timeslot-status.json');
+const TimeslotStatus  = require('../../constants/timeslotStatus.json');
```
Comment (catai9) [inline server/models/db/workerTimeslot.js:6]: Please note that the timeslot status constants file was renamed to lowerCamelCase to be consistent with the naming of the other constant files.

Comment (sehersarin) [reply server/models/db/workerTimeslot.js:6]: Good idea to keep naming consistent!

**Comment by sehersarin** [comment]
Hi Amy, great work on this PR. 

I have tried running the automated tests locally. 
- Inside the react-ui folder: All tests pass and for the AppointmentList.js, LogIn.js and CheckboxApplication.js components have 100% file coverage. 
- Inside the server folder when I run "npm test" I am getting some failed tests (could you please look into this - screenshot attached.) I have run the npm install command - is there anything else I should be doing?
Screen Shot 2020-11-13 at 11 57 21 AM

**Comment by catai9** [comment]
> 
> 
> Hi Amy, great work on this PR.
> 
> I have tried running the automated tests locally.
> 
>     * Inside the react-ui folder: All tests pass and for the AppointmentList.js, LogIn.js and CheckboxApplication.js components have 100% file coverage.
> 
>     * Inside the server folder when I run "npm test" I am getting some failed tests (could you please look into this - screenshot attached.)
> 
> 
> Screen Shot 2020-11-13 at 11 57 21 AM

Hi Seher,
Is it possible to provide more details on what errors thrown by the failing tests? Specifically, may you please screenshot the logs by searching for the FAIL test?

**Comment by sehersarin** [comment]
> > Hi Amy, great work on this PR.
> > I have tried running the automated tests locally.
> > ```
> > * Inside the react-ui folder: All tests pass and for the AppointmentList.js, LogIn.js and CheckboxApplication.js components have 100% file coverage.
> > 
> > * Inside the server folder when I run "npm test" I am getting some failed tests (could you please look into this - screenshot attached.)
> > ```
> > 
> > 
> > Screen Shot 2020-11-13 at 11 57 21 AM
> 
> Hi Seher,
> Is it possible to provide more details on what errors thrown by the failing tests? Specifically, may you please screenshot the logs by searching for the FAIL test?

This is the test that is failing for me (screenshot below.)

**Comment by sehersarin** [comment]
This is the test that is failing for me: getAppointments.test.js
Screen Shot 2020-11-13 at 11 50 44 AM

This image has more details of what I'm getting:
Screen Shot 2020-11-13 at 12 17 02 PM

**Comment by catai9** [comment]
Hi Seher, 
Please reference the bottom of #34. I will reiterate the key points below.

```
Yes, it appears that you have deleted some needed rows from your local database and added some unnecessary rows (outside of the scripts in the `migrations` folder). To address this issue, I would recommend creating a new database in pgAdmin (shown below) and then rerunning the code in the root folder using `npm run-script start:dev`. Afterwards, try doing `npm test` again in the react-ui and server folders and it should work. 

To create a new database in pgAdmin:
1. Right click on Databases and create a new database with a new name:
![image](https://user-images.githubusercontent.com/32720322/97991366-5eafe480-1daf-11eb-99a4-84a0de0d4c06.png)
2. Change the database name in the `.env` file and `.jest/setEnvVars.js`.
3. To avoid this problem in the future, it might be better to have one unique database for the testing and one that stores your local data. The reason it appears that multiple rows were inserted into the local database is because one of the automated unit tests attempts to insert rows. Since both the testing and local database point to the same location, the database becomes populated with test data.
```

Hope that helps!

**Comment by sehersarin** [comment]
Hi Amy, 

Perfect - got it up and running. Now all the server tests pass and have good code coverage. 
Screen Shot 2020-11-13 at 12 30 31 PM

**Comment by sehersarin** [comment]
Hello, 

I have finally completed testing including using Postman and testing the endpoint by sending requests to the api/cancel-specific-day endpoint on port 4000 with various combinations of query parameters. I have tested these examples provided, along with other test cases:

- workerId = 8000000 and specificDate = 2020-12-20 => true returned and relevant appointments marked as cancelled in local database table.
- no values specified for workerId or specificDate => error thrown
- invalid specificDate format, e.g., 'hello' => error thrown
- workerId is string => throw error

**Comment by sehersarin** [comment]
I am happy with this PR and am ready to approve and merge into the master branch!

**Review by sehersarin — APPROVED** [review]

**Pull request merged**